### PR TITLE
DEPR: DatetimeIndex setops with mismatched tzs

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -3089,14 +3088,10 @@ def _deprecate_dti_setop(self, other: Index, setop: str_t) -> None:
             and other.tz is not None
```
Comment (mroeschke) [inline pandas/core/indexes/base.py:3088]: Could also skip the `tz_convert` if both `.tz`s are "UTC"

Comment (jbrockmendel) [reply pandas/core/indexes/base.py:3088]: we only get here with non-matching dtypes.  i'll follow up to make that clearer

Comment (mroeschke) [reply pandas/core/indexes/base.py:3088]: Ah gotcha. That'd be helpful, thanks

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**